src/datepicker/calendar.js

```diff
@@ -112,8 +111,16 @@ export default class Calendar extends React.Component<
       this.focusCalendar();
     }
 
-    if (prevProps.value !== this.props.value) {
+    if (prevProps.value !== this.props.value && !this.state.focused) {
+      if (this.props.value instanceof Date) {
+        this.setState({highlightedDate: this.props.value});
+      } else if (this.props.value instanceof Array) {
+        this.setState({highlightedDate: this.props.value[0]});
+      } else {
+        this.setState({highlightedDate: new Date()});
+      }
```

Comment:
Changed this part to incorporate Nadia's suggestion. Added !this.state.focused to the conditional because without it highlight on click (specifically clicking to select range) is bugged. The second click to select an end day of range will render the endDate as blank (without the number inside rendered). !this.state.focused condition makes sure the reset occurs only when value is changed through external measures (like quickselect)